### PR TITLE
Cleanup nodeinfo.AttrTypeOSNameFull

### LINKED ISSUE #42: [Cleanup] Remove nodeinfo.AttrTypeOSNameFull
This attribute is redundant and can be derrived from `NodeLabelOSName` `NodeLabelOSVer`

### PR DESCRIPTION
**Pull request by Mmduh-483**
OSNameFull can be derrived from OSName and OSVer
Remove OSNameFull from Node Attributes package.

Fixes #42

### CODE REVIEW

File: pkg/nodeinfo/attributes_test.go [hunk 1]
```diff
@@ -72,22 +68,22 @@ var _ = Describe("NodeAttributes tests", func() {
 		})
 	})
 
-	Context("Create NodeAttributes from node with no OS name", func() {
+	Context("Create NodeAttributes from node with no OS version", func() {
```
Comment (adrianchiris) [inline pkg/nodeinfo/attributes_test.go:71]: Test case no longer relevant you can delete it

File: pkg/nodeinfo/attributes_test.go [hunk 2]
```diff
@@ -72,22 +68,22 @@ var _ = Describe("NodeAttributes tests", func() {
...
 			attr := newNodeAttributes(&testNode)
 
-			_, exist := attr.Attributes[AttrTypeOSNameFull]
+			_, exist := attr.Attributes[AttrTypeOSVer]
 			Expect(exist).To(BeFalse())
 		})
 	})
 
-	Context("Create NodeAttributes from node with no OS version", func() {
+	Context("Create NodeAttributes from node with no OS name", func() {
```
Comment (adrianchiris) [inline pkg/nodeinfo/attributes_test.go:81]: Test case no longer relevant you can delete it

File: pkg/state/state_nv_peer.go [hunk 3]
```diff
@@ -43,15 +43,9 @@ type stateNVPeer struct {
...
-type nvPeerRuntimeSpec struct {
-	CPUArch    string
-	OSNameFull string
-	Namespace  string
-}
-
 type nvPeerManifestRenderData struct {
 	CrSpec      *mellanoxv1alpha1.NVPeerDriverSpec
-	RuntimeSpec *nvPeerRuntimeSpec
+	RuntimeSpec *nodeInfoRuntimeSpec
```
Comment (adrianchiris) [inline pkg/state/state_nv_peer.go:48]: how is this change related to the cleanup?

File: pkg/state/state_skel.go [hunk 4]
```diff
@@ -34,6 +34,13 @@ import (
 	"github.com/Mellanox/network-operator/pkg/render"
 )
 
+type nodeInfoRuntimeSpec struct {
```
Comment (adrianchiris) [inline pkg/state/state_skel.go:37]: how is this change related to the cleanup?

**Review by adrianchiris — CHANGES_REQUESTED** [review]
Can you keep the changes to what was discussed in the issue.

i.e please dont rename members in the various state implementation and move to common struct even though they are not identical.

File: pkg/state/state_nv_peer.go (hunk 3 above)
Comment (Mmduh-483) [reply pkg/state/state_nv_peer.go:48]: it is refactoring for common attributes, you want it to be in a separated commit or PR?

File: pkg/state/state_skel.go (hunk 4 above)
Comment (Mmduh-483) [reply pkg/state/state_skel.go:37]: same above

File: pkg/state/state_nv_peer.go (hunk 3 above)
Comment (adrianchiris) [reply pkg/state/state_nv_peer.go:48]: i prefer to keep it as is was for now.

File: pkg/state/state_skel.go (hunk 4 above)
Comment (adrianchiris) [reply pkg/state/state_skel.go:37]: same as above

Comment (Mmduh-483) [reply pkg/state/state_skel.go]: updated

**Review by adrianchiris — APPROVED** [review]
